server/auth.js

```diff
@@ -42,6 +42,19 @@ internals.applyStrategy = function (server, next) {
                     }
 
                     done(null, Object.keys(results.user.roles));
+                }],
+                updateSession: ['scope', function (results, done) {
+
+                    if (!results.roles) {
```

Comment:
This should be checking for `if (!results.scope) {` since the scope and roles tasks run in parallel and we can't guarantee `results.roles` will be around at this point.

I noticed this in my testing after I realized the `lastActive` timestamp wasn't being updated.